torchao/experimental/quant_api.py

```diff
@@ -25,10 +25,16 @@
 logger.addHandler(handler)
 
 
-def _quantize(vals: torch.Tensor, group_size: int, nbit: int, has_weight_zeros: bool):
+def _quantize(vals: torch.Tensor, group_size: int, nbit: int, has_weight_zeros: bool, dtype=torch.int8):
     assert nbit >= 1 and nbit <= 8
-    qmin = -(1 << (nbit - 1))
-    qmax = (1 << (nbit - 1)) - 1
+    if dtype == torch.int8:
+        qmin = -(1 << (nbit - 1))
+        qmax = (1 << (nbit - 1)) - 1
+    elif dtype == torch.uint8:
+        qmin = 0
+        qmax = (1 << nbit) - 1
```

Comment:
well, `quantize_per_channel_group` in line 54, takes a `dtype` argument, not a `signed` argument.
If I do as you say, i would have to pass to `quantize_per_channel_group` the expression `signed ? dtype.int8 : dtype.uint8`
Why do that? Isn't it cleaner to just do what I did?

Reply:
I just meant that you used signed for your _quantize function and leave quantize_per_channel_group as is.

Why? Because right now all calls to `_quantize` now has to pass int8 or uint8 (depending on what the default value is), even when doing < 8 bit quant. So that feels a bit unintuitive. Instead if you just do `_quantize(vals, groups_size, nbit, has_weight_zero, signed=False)` then you know that you want to quantize using nbit and use unsigned type.

Reply:
ok, sounds good, coming right up. But i can't leave `quantize_per_channel_group` as is. I do need to pass `signed ? dtype.int8 : dtype.uint8` to it

Reply:
done!